academic-services/knowledge-exploration-service/reference-get-interpret.md

```diff
@@ -0,0 +1,459 @@
+`offset` | Optional | integer | The number of interpretations to skip in the result set. <br/><br/>Defaults to 0.
+`timeout` | Optional | integer | The maximum amount of time in milliseconds to use when generating interpretations. <br/><br/>If the timeout is hit, all interpretations that have been generated will be returned and a "timedOut" flag will be set in the response indicating that the timeout was hit before the total number of requested interpretations was met. <br/><br/>Defaults to 2000.
+`entityCount` | Optional | integer | The number of entities that should be returned for each interpretation that match the interpreted query expression. <br/><br/>Defaults to 0.
+`attributes` | Optional | string | A list of comma-separated attributes to include for each entity returned for each interpretation. See the [entity schema overview](reference-makes-api-entity-schema.md#entity-types) for the attributes that can be requested. <br/><br/>If no attributes are specified then the each entity will only contain its corresponding score. <br/><br/>If an asterisk (*) is specified, all available attributes will be returned. <br/><br/>Defaults to an empty string.
+
+## Responses
+
+Name | Type | Description
+--- | --- | ---
+200 OK | [InterpretResponse](#interpretresponse) | Interpret response successfully generated and returned.
+
+## Definitions
+
+| | |
+| --- | --- |
+[InterpretResponse](#interpretresponse) | Interpret response information.
+[Interpretation](#interpretation) | Information for an individual query interpretation.
+[InterpretationRuleMatch](#interpretationrulematch) | Information returned for grammar rules matched during interpretation.
+[InterpretationRuleMatchOutput](#interpretationrulematchoutput) | Output generated for a grammar rule match.
+
+### InterpretResponse
+
+Name | Type | Description
+--- | --- | ---
+interpretations | [Interpretation](#interpretation)[] | Array of interpretations ordered by relevance.
+timed_out_count | integer | The number of index partitions that timed out before returning the requested number of interpretations.
+string | The natural language query used to generate interpretations.
+
+### Interpretation
+
+Name | Type | Description
+--- | --- | ---
+logprob | double | The semantic score associated with the interpretation. See the [natural language queries](concepts-queries.md) page for more details.
+parse | string | The semantic parse of the natural language query. See the [natural language queries](concepts-queries.md) page for more details.
+rules | [InterpretationRuleMatch](#interpretationrulematch)[] | Array of grammar rules matched when generating interpretation.
+
+### InterpretationRuleMatch
+
+Name | Type | Description
+--- | --- | ---
+name | string | The name of the matching rule. For the default grammar this will always be "#GetPapers".
```

Comment:
>"#GetPapers" [](start = 91, length = 12)

nit: can we link to the list of rules available?

Reply:
Rules are defined in the grammar itself, and for the default grammar this is the only rule.

---
In reply to: [373237870](https://github.com/MicrosoftDocs/microsoft-academic-services/pull/97#discussion_r373237870) [](ancestors = 373237870)